### PR TITLE
[IMP] website_event[|track|exhibitor]: improve UI of event app

### PR DESCRIPTION
**Pull request by pyduf**
Improve UI of event app.

Detailed changes:

1) change the wording of the selection item "sponsor" of
event_sponsor.exhibitor_type from "sponsor" to "Footer Logo Only"
2) add a placeholder for the name in the form when creating an event page
3) in the track page and exhibitor page, if the description is empty:
- for external user: hide the empty description zone for external user
- for internal user: display an alert (info) inviting the user to complete it
with a link to the backend
3 bis) in the track page and exhibitor page, the page and the menu aside
are glued to the top. Add some space there.
4) display unpublished tag on the side menu for unpublished tracks

Some other changes:
- when creating an event in the front, add a red border on event name input
when submitting an empty field
- When some content are empty, there were some awkward horizontal lines
separating nothing, we have decided to remove them
- Fix track layout and improve mobile layout
- Fix event track starts warning display condition

Technical remarks:

1) the change is done on the model and then affect view globally
(ex.: also "card view")

Task-2692907

### CODE REVIEW

File: addons/website_event/static/src/xml/event_create.xml [hunk 1]
```diff
@@ -5,7 +5,7 @@
             <div class="form-group row" id="name">
                 <label for="name" class="col-md-2 col-form-label">Event Name</label>
                 <div class="col-md-10">
-                    <input type="text" name="name" class="form-control" required="required"/>
+                    <input type="text" name="name" class="form-control" required="required" placeholder="e.g. Conference for Architects"/>
```
Comment (tde-banana-odoo) [inline addons/website_event/static/src/xml/event_create.xml:8]: Linting: could be multi-lined as it creates long lines. We don't have hard rules for line length, we try to keep it somehow around max 100-120 (depending on what remains), aka
```
<input type="text" name="name" class="form-control" required="required"
       placeholder="e.g. Conference for Architects"/>
```

File: addons/website_event_track/views/event_track_templates_page.xml [hunk 2]
```diff
@@ -128,7 +132,15 @@
                 <div t-field="track.partner_biography" class="oe_no_empty"/>
                 <hr t-if="not is_html_empty(track.description)" class="mt-2 pb-1 mb-1"/>
             </div>
-            <div t-field="track.description" class="my-2 mx-3 oe_no_empty"/>
+
+            <!-- Description -->
+            <t>
+                <div t-if="not is_html_empty(track.description)" t-field="track.description" class="my-2 mx-3 oe_no_empty"/>
+                <div t-elif="not env.user.share" class="my-2 mx-3 oe_no_empty">
```
Comment (tde-banana-odoo) [inline addons/website_event_track/views/event_track_templates_page.xml:139]: Actually I think we should ensure can effectively edit tracks. We could probably simply check he is event officer (having 'event.group_user'). Otherwise internal users will not be able to add a description as they have no rights to edit tracks.

File: addons/website_event_track/views/event_track_templates_page.xml [hunk 3]
```diff
@@ -202,7 +214,7 @@
     <div class="d-flex align-items-center">
         <small class="text-muted" t-esc="track_other.partner_name"/>
         <div class="d-inline-block ml-auto o_wesession_track_aside_info text-truncate">
-            <small t-if="not track_other.website_published and user_event_manager"
+            <small t-if="not track_other.website_published and env.user.has_group('event.group_event_manager')"
```
Comment (tde-banana-odoo) [inline addons/website_event_track/views/event_track_templates_page.xml:217]: Why this change ? Is the "user_event_manager" variable invalid ?

**Review by tde-banana-odoo — COMMENTED** [review]
Early validation (except the 'mode' content, have to dig a bit)

File: addons/website_event_exhibitor/views/event_exhibitor_templates_page.xml [hunk 4]
```diff
@@ -30,6 +32,7 @@
 <!-- ============================================================ -->
 
 <template id="exhibitor_main" name="Exhibitor: Main Content">
+    <t t-set="has_sponsors_other" t-value="has_sponsors_other if has_sponsors_other is not None else True"/>
```
Comment (tde-banana-odoo) [inline addons/website_event_exhibitor/views/event_exhibitor_templates_page.xml:35]: Does not seems used anymore ?

File: addons/website_event_track/views/event_track_templates_agenda.xml [hunk 5]
```diff
@@ -4,7 +4,20 @@
 <!-- Revamped agenda : Will need to replace agenda_online from website_event_track in master -->
 <template id="agenda_online" name="Track Online: Agenda">
     <t t-call="website_event.layout">
-        <div class="o_wevent_online o_weagenda_index">
+        <!-- No tracks -->
+        <div class="container" t-if="not tracks_by_days">
```
Comment (tde-banana-odoo) [inline addons/website_event_track/views/event_track_templates_agenda.xml:8]: Just for styling, I would put the t-if attribute first (ease to understand code flow when reading template), like
```
<div t-if="blah" class="..">
    ...
</div>
<div t-else="" class="...">
    ...
</div>
```

File: addons/website_event_track/views/event_track_templates_agenda.xml [hunk 6]
```diff
@@ -59,20 +72,7 @@
...
-        </div>
-        <div t-else="" class="alert alert-info text-center" groups="event.group_event_user">
-            <a target="_blank" t-att-href="'/web?#action=website_event_track.action_event_track_from_event&amp;active_id=%s' % event.id">
-                <p class="m-0">Schedule some tracks to get started</p>
-            </a>
-        </div>
-    </div>
-
-    <section t-else="" class="col-12" t-foreach="days" t-as="day">
+    <section t-if="tracks_by_days" class="col-12" t-foreach="days" t-as="day">
```
Comment (tde-banana-odoo) [inline addons/website_event_track/views/event_track_templates_agenda.xml:75]: As anyway it is called by parent template only when having tracks_by_days, I would remove this if. That way it is now the caller's job to ensure it has values to display before calling this template.

File: addons/website_event_track/views/event_track_templates_page.xml [hunk 7]
```diff
@@ -29,6 +31,7 @@
 <!-- ============================================================ -->
 
 <template id="event_track_content" name="Track: Main Description">
+    <t t-set="has_tracks_other" t-value="has_tracks_other if has_tracks_other is not None else True"/>
```
Comment (tde-banana-odoo) [inline addons/website_event_track/views/event_track_templates_page.xml:34]: Does not seems used anymore ?

File: addons/website_event_track/views/event_track_templates_page.xml (hunk 3 above)
Comment (tde-banana-odoo) [reply addons/website_event_track/views/event_track_templates_page.xml:214]: Reading again, seems like a fix we should backport to stable, isn't it ? Variable user_event_manager does not seem to exist anymore.

File: addons/website_event_track/views/event_track_templates_page.xml [hunk 8]
```diff
@@ -128,7 +129,15 @@
                 <div t-field="track.partner_biography" class="oe_no_empty"/>
                 <hr t-if="not is_html_empty(track.description)" class="mt-2 pb-1 mb-1"/>
             </div>
-            <div t-field="track.description" class="my-2 mx-3 oe_no_empty"/>
+
+            <!-- Description -->
+            <t>
+                <div t-if="not is_html_empty(track.description)" t-field="track.description" class="my-2 mx-3 oe_no_empty"/>
+                <div t-elif="env.user.has_group('event.group_event_user')" class="alert alert-info mx-3 mt-3">
+                    This Track does not have a description.
```
Comment (tde-banana-odoo) [inline addons/website_event_track/views/event_track_templates_page.xml:137]: Do we need an uppercase on track ?

**Review by tde-banana-odoo — COMMENTED** [review]
Validation

**Comment by pyduf** [comment]
Hello @tde-banana-odoo, 

Thanks for all the relevant comments. I have corrected everything except the point: 
"Reading again, seems like a fix we should backport to stable, isn't it ? Variable user_event_manager does not seem to exist anymore."

The line was added in the commit #7ff89c18 (24/05/2021) but when I checkout the project at that commit, there is no trace of the variable "user_event_manager" except on that line. 
So I suppose that the problem was already there when adding that functionality to display unpublished: the condition is always false. 
So the functionality of displaying 'unpublished" for track unpublished has never been active.

I suppose the condition is present to display a warning to the event manager only.

Is it worth to implement that warning in the past ? 
If so I suppose that it was decided to implement it on 24/05/2021, we can target version 15 (which was released 5 month later) ?

Thanks

**Comment by tde-banana-odoo** [comment]
Ok to backport in v15 then :)

**Review by tde-banana-odoo — COMMENTED** [review]
Validation: LGTM, one small comment :)

File: addons/website_event_track/views/event_track_templates_page.xml [hunk 9]
```diff
@@ -99,37 +117,39 @@
...
-                        <span class="mb-1 d-flex align-items-baseline text-break" t-if="track.partner_id.website">
-                            <i class="fa fa-home mr-2"/><a t-att-href="track.partner_id.website"><span t-field="track.partner_id.website"/></a>
-                        </span>
-                        <span class="mb-1 d-flex align-items-baseline text-break" t-if="track.partner_email">
-                            <i class="fa fa-envelope mr-2"/><a t-att-mailto="track.partner_email"><span t-field="track.partner_email"/></a>
-                        </span>
-                        <span class="mb-1 d-flex align-items-baseline text-break" t-if="track.partner_phone">
-                            <i class="fa fa-phone mr-2"/><span t-field="track.partner_phone"/>
-                        </span>
+            <t t-if="track.image or track.partner_name or track.partner_function or track.partner_id.website or track.partner_email or track.partner_phone or not is_html_empty(track.partner_biography)">
```
Comment (awa-odoo) [inline addons/website_event_track/views/event_track_templates_page.xml:122]: Sugg to make it a bit more readable:

```xml
<t t-set="track_fields" t-value="['image', 'partner_name', 'partner_function', 'partner_email', 'partner_phone']"/>
<!-- Avoid rendering a section for track information if we don't have anything to display -->
<t t-if="any(track[track_field] for track_field in track_fields) or track.partner_id.website or not is_html_empty(track.partner_biography)"/>
```

File: addons/website_event_track/views/event_track_templates_page.xml [hunk 10]
```diff
@@ -99,37 +117,39 @@
...
+                    <div t-if="not is_html_empty(track.partner_biography)" t-field="track.partner_biography"
+                         class="oe_no_empty"/>
+                    <hr t-if="not is_html_empty(track.description)" class="mt-2 pb-1 mb-1"/>
                 </div>
-                <div t-field="track.partner_biography" class="oe_no_empty"/>
-                <hr t-if="not is_html_empty(track.description)" class="mt-2 pb-1 mb-1"/>
+            </t>
+
+            <!-- Description -->
+            <div t-if="not is_html_empty(track.description)" t-field="track.description" class="my-2 mx-3 oe_no_empty"/>
```
Comment (awa-odoo) [inline addons/website_event_track/views/event_track_templates_page.xml:145]: If you only have a "track.description" and nothing else, then you will not have any "hr" tag to split I think, as you will not enter the block here above.

Not sure if intended or not, maybe to double-check.

File: addons/website_event_track/views/event_track_templates_page.xml [hunk 11]
```diff
@@ -99,37 +117,39 @@
...
+            <div t-elif="env.user.has_group('event.group_event_user')" class="alert alert-info mx-3 mt-3">
```
Comment (awa-odoo) [inline addons/website_event_track/views/event_track_templates_page.xml:146]: That is strange, as it redirects you to the backend form view while you should be able to write a description using the website editor?

**Review by awa-odoo — COMMENTED** [review]
Hello @pyduf 

Finally had a look at this one, sorry for the delay!
Added a few questions / suggestions.

You also might want to rebase to freshen up the branch a bit.

Cheers!

**Comment by awa-odoo** [comment]
@robodoo r+
